challenge-11/challenge-11.js

```diff
@@ -1,54 +1,89 @@
+	'Entrou ao menos uma vez!'
+	Qual loop você deve usar para que essa mensagem seja mostrada no console?
+	*/
+	var once = false;
+	do{
+		console.log( 'Entrou ao menos uma vez!');
+	} 
+	while( once === true );
 
-/*
-Crie um objeto chamado `person`, que receba as seguintes propriedades:
-- 'name', 'age', 'weight' e 'birthday'. Preencha com os valores corretos
-para o nome, idade, peso e data de nascimento dessa pessoa.
-*/
-// ?
+	/*
+	Crie um objeto chamado `person`, que receba as seguintes propriedades:
+	- 'name', 'age', 'weight' e 'birthday'. Preencha com os valores corretos
+	para o nome, idade, peso e data de nascimento dessa pessoa.
+	*/
+	var person = {
+		name: 'Anderson',
+		age: 19,
+		weight: '94 Kg',
+		birthday: '27/08/1995'
+	};
 
-/*
-Use um loop para percorrer o objeto criado acima, mostrando no console
-a frase:
-'The [PROPERTY] of person is [VALUE]'
-Aproveite e crie uma variável `counter` que vai contar quantas propriedades
-esse objeto tem.
-Após o loop, mostre a frase:
-'The person has [COUNTER] properties'
-*/
-// ?
+	/*
+	Use um loop para percorrer o objeto criado acima, mostrando no console
+	a frase:
+	'The [PROPERTY] of person is [VALUE]'
+	Aproveite e crie uma variável `counter` que vai contar quantas propriedades
+	esse objeto tem.
+	Após o loop, mostre a frase:
+	'The person has [COUNTER] properties'
+	*/
+	for( prop in person){
+		console.log('The ' + prop + ' of person is ' + person[prop] + ' ');
+	}
+	console.log('The person has ' + prop.length + ' properties');
 
-/*
-Crie uma função chamada `moreThan`, que vai verificar se a pessoa (objeto
-criado acima) é mais velha que a idade passada por parâmetro.
-Se verdadeiro, retornar `true`. Senão, retornar false.
-Após a função, mostrar a mensagem no console:
-'The person has more than 25 years old? [TRUE/FALSE]'
-*/
-// ?
+	/*
+	Crie uma função chamada `moreThan`, que vai verificar se a pessoa (objeto
+	criado acima) é mais velha que a idade passada por parâmetro.
+	Se verdadeiro, retornar `true`. Senão, retornar false.
+	Após a função, mostrar a mensagem no console:
+	'The person has more than 25 years old? [TRUE/FALSE]'
+	*/
+	function moreThan(age){
+		if( age > person.age ){
+			return true;
+		}
+		return false;
+	}
 
-/*
-Faça um loop de 0 a 20, que adicione cada número como um item de um
-array chamado `numbers`. Se o contador for maior que 10, saia do loop.
-Mostre no console os números no array.
-*/
-console.log( 'De 0 a 10:' );
-// ?
-
-/*
-Faça outro loop de 0 a 20, que adicione a um array chamado `numbers` (já
-criado acima, só precisa ser reiniciado) números de 0 a 20, inclusive
-esses. Se o número for ímpar, pular para o próximo número.
-Mostrar no console os números do array.
-*/
-console.log( 'Pares de 0 a 20:' );
-// ?
+	/*
+	Faça um loop de 0 a 20, que adicione cada número como um item de um
+	array chamado `numbers`. Se o contador for maior que 10, saia do loop.
+	Mostre no console os números no array.
+	*/
+	console.log( 'De 0 a 10:' );
```

Comment:
Não faz muita diferença. A diferença não está na iteração, mas sim na forma como você trata os dados. Para tudo o que for estático, se você salvar em variáveis antes da iteração, o loop vai ficar mais rápido ;)